ambry-server/src/integration-test/java/com.github.ambry.server/ServerTestUtil.java

```diff
@@ -270,16 +286,48 @@ protected static void endToEndTest(Port targetPort, String routerDatacenter, Str
       // verify user metadata
       ByteBuffer userMetadataOutput = MessageFormatRecord.deserializeUserMetadata(responseStream);
       Assert.assertArrayEquals(usermetadata, userMetadataOutput.array());
+      if (testEncryption) {
+        assertNotNull("MessageMetadata should not have been null",
+            resp3.getPartitionResponseInfoList().get(0).getMessageMetadataList().get(0));
+        assertArrayEquals("EncryptionKey mismatch", encryptionKey,
+            resp3.getPartitionResponseInfoList().get(0).getMessageMetadataList().get(0).getEncryptionKey().array());
+      } else {
+        assertNull("MessageMetadata should have been null",
+            resp3.getPartitionResponseInfoList().get(0).getMessageMetadataList().get(0));
+      }
+
+      // get blob all
+      GetRequest getRequest4 =
+          new GetRequest(1, "clientid2", MessageFormatFlags.All, partitionRequestInfoList, GetOption.None);
+      channel.send(getRequest4);
+      stream = channel.receive().getInputStream();
+      GetResponse resp4 = GetResponse.readFrom(new DataInputStream(stream), clusterMap);
+      responseStream = resp4.getInputStream();
+      BlobAll blobAll = MessageFormatRecord.deserializeBlobAll(responseStream, blobIdFactory);
+      byte[] actualBlobData = new byte[(int) blobAll.getBlobData().getSize()];
+      blobAll.getBlobData().getStream().getByteBuffer().get(actualBlobData);
+      // verify content
+      Assert.assertArrayEquals("Content mismatch", data, actualBlobData);
+      if (testEncryption) {
+        Assert.assertNotNull("EncryptionKey should not ne null", blobAll.getBlobEncryptionKey());
+        Assert.assertArrayEquals("EncryptionKey mismatch", encryptionKey, blobAll.getBlobEncryptionKey().array());
+      } else {
+        Assert.assertNull("EncryptionKey should have been null", blobAll.getBlobEncryptionKey());
+      }
 
```

Comment:
see if it is easily possible to do a `GetRequest()` here for multiple blobs that include a mix of encrypted and non-encrypted ones.

Reply:
As of now I have parametrized the method and hence it tests only only flow at a time. Will add more in my next patch.